robottelo/decorators/func_locker.py

```diff
@@ -0,0 +1,165 @@
+# -*- encoding: utf-8 -*-
+"""Implements test function locking, using pytest_services file locking
+
+Usage::
+
+
+    from robottelo.decorators.func_locker import (
+        locking_function,
+        lock_function,
+     )
+
+    # in many cases we have tests that need some test functions to run isolated
+    # from other other py.test workers when used in --boxed mode
+    class SomeTestCase(TestCase):
+
+        @classmethod
+        @lock_function
+        def setUpClass(cls):
+            pass
+
+    # in other cases we want only a portion of the test function to be isolated
+    class SomeTestCase(TestCase):
+
+        @classmethod
+        def setUpClass(cls):
+
+            with locking_function(cls.setUpClass, 'publish_puppet_class'):
+                # call the publish function
+
+
+    # some tests can be in conflicts with other tests parts
+    class SomeTestCase(TestCase):
+
+       @lock_function
+       def test_to_lock(self):
+          pass
+
+       def test_that_conflict_with_test_to_lock(self)
+            with locking_function(self.test_to_lock):
+                # do some operations that conflict with test_to_lock
+"""
+import functools
+import logging
+import os
+import tempfile
+import unittest2
+
+from contextlib import contextmanager
+
+from pytest_services.locks import file_lock
+
+logger = logging.Logger(__name__)
+
+TEMP_ROOT_DIR = 'robottelo'
+TEMP_FUNC_LOCK_DIR = 'lock_functions'
+LOCK_DIR = None
+
+
+def _get_temp_lock_function_dir(create=True):
+    global LOCK_DIR
+    if LOCK_DIR is not None:
+        return LOCK_DIR
+    tmp_lock_dir = os.path.join(tempfile.gettempdir(), TEMP_ROOT_DIR,
+                                TEMP_FUNC_LOCK_DIR)
+    if create and not os.path.exists(tmp_lock_dir):
+        os.makedirs(tmp_lock_dir)
+
+    LOCK_DIR = tmp_lock_dir
+
+    return LOCK_DIR
+
+
+def _get_function_name(function, context=None):
```

Comment:
We already have a function for this purpose in https://github.com/SatelliteQE/robottelo/blob/6.2.z/robottelo/helpers.py#L413

do you think we can extend the existing function to serve all the purposes?